src/bci_build/package/appcontainers.py

```diff
@@ -249,6 +249,93 @@
         )
     )
 
+
+_POSTFIX_FILES = {}
+for filename in (
+    "entrypoint.sh",
+    "entrypoint.sles.sh",
+    "postfix-manpages.md",
+    "smtpd_sender_login_maps",
+    "virtual_alias_domains",
+    "virtual_alias_maps",
+    "virtual_gid_maps",
+    "virtual_mailbox_maps",
+    "virtual_uid_maps",
+):
+    if filename.startswith("entrypoint.") or filename.endswith(".md"):
+        if os_version != OsVersion.TUMBLEWEED and filename == "entrypoint.sles.sh":
+            _POSTFIX_FILES["entrypoint.sh"] = (
+                Path(__file__).parent / "postfix" / "entrypoint" / filename
+            ).read_bytes()
+        else:
+            _POSTFIX_FILES[filename] = (
+                Path(__file__).parent / "postfix" / "entrypoint" / filename
+            ).read_bytes()
+    else:
+        _POSTFIX_FILES[filename] = (
+            Path(__file__).parent / "postfix" / "entrypoint" / "ldap" / filename
+        ).read_bytes()
+
+
+POSTFIX_CONTAINERS = [
+    ApplicationStackContainer(
+        name="postfix",
+        package_name="postfix-image",
+        pretty_name="Postfix",
+        custom_description="Postfix container is fast and secure mail server, {based_on_container}.",
+        os_version=os_version,
+        is_latest=os_version in CAN_BE_LATEST_OS_VERSION,
+        version="%%postfix_version%%",
+        version_in_uid=False,
+        replacements_via_service=[
+            Replacement(
+                regex_in_build_description="%%postfix_version%%",
+                package_name="postfix",
+                parse_version="minor",
+            )
+        ],
+        package_list=[
+            name
+            for name in (
+                "ca-certificates-mozilla",
+                "cyrus-sasl",
+                "cyrus-sasl-plain",
+                "ed",
+                "netcfg",
+                "postfix",
+                "postfix-ldap",
+                "postfix-lmdb",
+                "timezone",
+            )
+            + (
+                (
+                    "mandoc",
+                    "spamassassin-spamc",
+                    "spamass-milter",
+                )
+                if os_version == OsVersion.TUMBLEWEED
+                else ()
+            )
+        ],
+        entrypoint=["/entrypoint/entrypoint.sh && cat /entrypoint/postfix-manpages.md"],
+        cmd=["postfix", "start"],
+        extra_files=_POSTFIX_FILES,
+        support_level=SupportLevel.TECHPREVIEW,
+        exposes_tcp=[25, 465, 587],
+        volumes=["/var/spool/postfix", "/var/spool/vmail", "/etc/pki"],
+        custom_end=f"""{DOCKERFILE_RUN} mkdir -p /entrypoint/ldap
+COPY entrypoint.sh postfix-manpages.md /entrypoint/
+{DOCKERFILE_RUN} chmod +x /entrypoint/entrypoint.sh
+COPY smtpd_sender_login_maps virtual_alias_domains virtual_alias_maps virtual_gid_maps virtual_mailbox_maps virtual_uid_maps /entrypoint/ldap/
+
+HEALTHCHECK --interval=5s --timeout=10s --start-period=30s --retries=3 \
+        CMD postfix status
```

Comment:
Does this work for you? When I try the built container, then it never becomes healthy as `postfix status` reports that the service is not running

```suggestion
HEALTHCHECK --interval=5s --timeout=10s --start-period=30s --retries=3 \
        CMD postfix status
```

Reply:
@dcermak, locally, I've tried building the container image with a `--format docker` flag to buildah.
Without the flag, there is no HEALTHCHECK created.

`WARN[0177] HEALTHCHECK is not supported for OCI image format and will be ignored. Must use `docker` format`

---

And with the recent postfix version bump to 3.9.x in Factory, I do need to update the ldap maps, for the below warnings. But the container gets healthy after 1/2 tries.


```
> sudo buildah bud --format docker -t postfix-image .

> sudo podman run --rm -it --name postfix -p 8488:25  -e SERVER_HOSTNAME=localhost postfix-image

> sudo podman inspect postfix --format "{{json .State.Health}}" | jq
{
  "Status": "healthy",
  "FailingStreak": 0,
  "Log": [
    {
      "Start": "2024-05-29T12:06:19.0453439+05:30",
      "End": "2024-05-29T12:06:20.079235802+05:30",
      "ExitCode": 1,
      "Output": "postfix: fatal: bad string length 0 < 1: myhostname = "
    },
    {
      "Start": "2024-05-29T12:06:50.882187668+05:30",
      "End": "2024-05-29T12:06:50.928231067+05:30",
      "ExitCode": 0,
      "Output": "/usr/sbin/postconf: warning: /etc/postfix/main.cf: support for parameter \"smtp_use_tls\" will be removed; instead, specify \"smtp_tls_security_level\"\n/usr/sbin/postconf: warning: /etc/postfix/main.cf: support for parameter \"smtpd_enforce_tls\" will be removed; instead, specify \"smtpd_tls_security_level\"\n/usr/sbin/postconf: warning: /etc/postfix/main.cf: support for parameter \"smtpd_use_tls\" will be removed; instead, specify \"smtpd_tls_security_level\"\n/usr/sbin/postconf: warning: /etc/postfix/main.c"
    }
  ]
}

```

Reply:
Please try out the image from OBS. The bot created a test build and added registry links in the comments, e.g. here: https://github.com/SUSE/BCI-dockerfile-generator/pull/1046#issuecomment-2134576192

Unfortunately OBS builds images differently than a plain `buildah bud` or `docker build`.

Reply:
Oh, I totally missed the registry links for staging images all this time. Thanks for pointing. 

Testing again!

Reply:
@dcermak, after the fix in https://github.com/SUSE/BCI-dockerfile-generator/pull/1046#discussion_r1618291096, the HEALTHCHECK in the built image also works fine for me.

I followed:

```
> sudo podman run --rm -it --name postfix -p 8684:25 -e SERVER_HOSTNAME=localhost registry.opensuse.org/home/defolos/bci/staging/tumbleweed/tumbleweed-1046/containerfile/opensuse/postfix

> podman inspect postfix --format "{{json .State.Health}}" | jq
{
  "Status": "healthy",
  "FailingStreak": 0,
  "Log": [
    {
      "Start": "2024-05-29T21:04:45.362446147+05:30",
      "End": "2024-05-29T21:04:45.418870472+05:30",
      "ExitCode": 0,
      "Output": "/usr/sbin/postconf: warning: /etc/postfix/main.cf: support for parameter \"smtp_use_tls\" will be removed; instead, specify \"smtp_tls_security_level\"\n/usr/sbin/postconf: warning: /etc/postfix/main.cf: support for parameter \"smtp_enforce_tls\" will be removed; instead, specify \"smtp_tls_security_level\"\n/usr/sbin/postconf: warning: /etc/postfix/main.cf: support for parameter \"smtpd_use_tls\" will be removed; instead, specify \"smtpd_tls_security_level\"\n/usr/sbin/postconf: warning: /etc/postfix/main.cf:"
    },
   ...
   ...
   ]
```